src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -173,6 +174,33 @@ public function loadClass($class)
                 @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
             }
 
+            // Inherit @final annotations
+            self::$finalMethods[$name] = $parent && isset(self::$finalMethods[$parent]) ? self::$finalMethods[$parent] : array();
+
+            foreach ($refl->getMethods(\ReflectionMethod::IS_PUBLIC | \ReflectionMethod::IS_PROTECTED) as $method) {
+                // It's not the declaring class
+                if ($method->class !== $name) {
+                    continue;
+                }
+
+                if ($parent && isset(self::$finalMethods[$parent][$method->name])) {
+                    // Fetch the real class declaring the parent method
+                    $prototype = $method->getPrototype();
```

Comment:
I guess you don't need this call if you compute the message whe populating `self::$finalMethods[$name][$method->name]`?